src/transport/encoding/raptorq/decoder.rs

```diff
@@ -13,17 +13,19 @@ use raptorq::{Decoder as ExtDecoder, EncodingPacket};
 use serde::{Deserialize, Serialize};
 use tracing::{trace, warn};
 
-use super::ChunkedPayload;
+use super::{ChunkedPayload, TRANSMISSION_INFO_SIZE, UID_SIZE};
 use crate::encoding::message::Message;
 use crate::encoding::payload::BroadcastPayload;
 use crate::transport::encoding::Configurable;
 use crate::transport::Decoder;
 
-const DEFAULT_CACHE_TTL_SECS: u64 = 60;
-const DEFAULT_CACHE_PRUNE_EVERY_SECS: u64 = 60 * 5;
+const DEFAULT_CACHE_TTL: Duration = Duration::from_secs(60);
+const DEFAULT_CACHE_PRUNE_EVERY: Duration = Duration::from_secs(30);
+
+const DEFAULT_MAX_UDP_LEN: u64 = 10 * 1_024 * 1_024;
 
 pub struct RaptorQDecoder {
-    cache: HashMap<[u8; 32], CacheStatus>,
+    cache: BTreeMap<[u8; UID_SIZE + TRANSMISSION_INFO_SIZE], CacheStatus>,
```

Comment:
Why the change to a `BTreeMap`? Do you need ordering?

Reply:
It's part of an audit finding.

Basically it's related to the `retain` function and its performance

_"In the current implementation, this operation takes O(capacity) time instead of O(len) because it internally visits empty buckets too."_ ([src](https://doc.rust-lang.org/std/collections/struct.HashMap.html#performance-7))